src/Ledger/TokenAlgebra/ValueVector.agda

```diff
@@ -0,0 +1,50 @@
+{-# OPTIONS --safe #-}
+open import Ledger.Prelude hiding (_∙_; ε)
+
+module Ledger.TokenAlgebra.ValueVector (PolicyId : Type) (n : ℕ) where
+
+import Algebra as Alg
+open import Algebra.Morphism.Construct.DirectProduct.Ext
+open import Algebra.Construct.DirectProduct
+open import Data.Nat.Properties using (+-0-commutativeMonoid)
+import Data.Product.Relation.Binary.Pointwise.NonDependent as Product
+open import Data.Vec as Vec
+  hiding (fromList)
+import Data.Vec.Instances.Ext as Vec
+import Data.Vec.Relation.Binary.Pointwise.Inductive as Vec
+
+open import Ledger.TokenAlgebra PolicyId
+
+Quantity = ℕ
+
+Vec-commutativeMonoid = Vec.commutativeMonoid n +-0-commutativeMonoid
+
+module +-0-commutativeMonoid = Alg.CommutativeMonoid +-0-commutativeMonoid
+module Vec-commutativeMonoid = Alg.CommutativeMonoid Vec-commutativeMonoid
+
+module _ (Policies : Vec PolicyId n) where
+  Value-TokenAlgebra : TokenAlgebra
```

Comment:
I think it would be good to also implement the FFI for this type in this PR, since that seems to be the main point of this particular algebra. I'd say it should map to a `Coin × List (PolicyId × String × ℕ)`, and the conversion to the Agda type should error if any of the `String`s are nonempty or there are any duplicate `PolicyId`s.